src/main/java/uk/gov/companieshouse/company_appointments/mapper/ResourceChangedRequestMapper.java

```diff
@@ -0,0 +1,34 @@
+package uk.gov.companieshouse.company_appointments.mapper;
+
+import java.util.function.Supplier;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.chskafka.ChangedResource;
+import uk.gov.companieshouse.api.chskafka.ChangedResourceEvent;
+import uk.gov.companieshouse.company_appointments.model.data.ResourceChangedRequest;
+
+@Component
+public class ResourceChangedRequestMapper {
+
+    private final Supplier<String> timestampGenerator;
+
+    public ResourceChangedRequestMapper(Supplier<String> timestampGenerator) {
+        this.timestampGenerator = timestampGenerator;
+    }
+
+    public ChangedResource mapChangedResource(ResourceChangedRequest request) {
+        ChangedResourceEvent event = new ChangedResourceEvent().publishedAt(this.timestampGenerator.get());
+        ChangedResource changedResource = new ChangedResource()
+                .resourceUri(String.format("company/%s/officers", request.getCompanyNumber()))
```

Comment:
the resource URI should contain the appointmentId for the appointment which has changed

Reply:
Resolved in commit: https://github.com/companieshouse/company-appointments.api.ch.gov.uk/pull/80/commits/e2635f2cbba4f7401f8de2a618f1b5c29b21ed73